phoenix-core/src/test/java/org/apache/phoenix/parse/QueryParserTest.java

```diff
@@ -925,6 +925,31 @@ public void testShowCreateTable() throws Exception {
         // Expected failures.
         parseQueryThatShouldFail("SHOW CREATE VIEW foo");
         parseQueryThatShouldFail("SHOW CREATE TABLE 'foo'");
+    }
+
+    @Test
+    public void testBinaryLiteral() throws Exception {
+        // Happy paths
+        parseQuery("SELECT b, x from x WHERE x = x'00'");
+        parseQuery("SELECT b, x from x WHERE x = "
+                + "x  '0 12 ' --comment \n /* comment */ '34 567' \n \n 'aA'");
+        parseQuery("SELECT b, x from x WHERE x = "
+                + "b  '0 10 ' --comment \n /* comment */ '10 101' \n \n '00000000'");
 
+        // Expected failures.
```

Comment:
Some comments here would be welcome. In particular I'm having trouble seeing why 934 should pass and 940 should fail. (I also find it surprising that 934 and 936 should pass given the use of whitespace in the WHERE clause.)

Reply:
Thanks.
I've misread the grammar in the standard, space is allowed before and after digits, but not between X and ' , as you would expect.